client/src/components/DatasetMoveSamplesModal.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { Box, Grid, RadioButton, Text } from 'grommet'
+import { config } from 'config'
+import { useResponsive } from 'hooks/useResponsive'
+import { Button } from 'components/Button'
+import { Modal, ModalBody } from 'components/Modal'
+import { WarningText } from 'components/WarningText'
+import { InfoText } from 'components/InfoText'
+import { Link } from 'components/Link'
+
+export const DatasetMoveSamplesModal = ({
+  label = 'Move Samples',
+  title = 'Move Samples to My Dataset',
+  modalDisabled = false,
+  appendDisabled = false
+}) => {
+  const [showing, setShowing] = useState(false)
+  const [checkedAppendSamples, setCheckedAppendSamples] = useState(false)
+  const [checkedReplaceSamples, setCheckedReplaceSamples] = useState(false)
```

Comment:
With `RadioButtonGroup`, we can define the radio button options and manage the `value` change using a single state.

For example:

```suggestion
 
  const radioOptions = [
    { label: 'Append samples to My Dataset', value: 'append' },
    { label: 'Replace samples in My Dataset', value: 'replace' }
  ]
  const [action, setAction] = useState(radioOptions[0].value) // set default as 'append' here  
```